test-butler-app/src/main/java/com/linkedin/android/testbutler/ButlerService.java

```diff
@@ -167,6 +175,9 @@ public void onDestroy() {
 
         // Restore the original keyboard setting
         showImeWithHardKeyboardHelper.restoreShowImeState(getContentResolver());
+
+        // Re-enable immersive mode confirmation
+        immersiveModeDialogDisabler.setState(getContentResolver(), true);
```

Comment:
We should be setting this back to the original value, not `true` (it might have been false before the test started, in which case this is a destructive action)